packages/data-driver-postgres/src/query/json.ts

```diff
@@ -0,0 +1,13 @@
+import {
+	tableIndexToIdentifier,
+	type AbstractSqlQuerySelectJsonNode,
+	columnIndexToIdentifier,
+} from '@directus/data-sql';
+import { escapeIdentifier } from '../utils/escape-identifier.js';
+
+export function json(node: AbstractSqlQuerySelectJsonNode): string {
+	const tableAlias = tableIndexToIdentifier(node.tableIndex);
+	const columnAlias = columnIndexToIdentifier(node.columnIndex);
+	const column = `${escapeIdentifier(tableAlias)}.${escapeIdentifier(node.columnName)}`;
+	return `${column} -> ${node.path.map((i) => escapeIdentifier(i)).join(` ->> `)} AS ${escapeIdentifier(columnAlias)}`;
```

Comment:
```escapeIdentifier``` is not the correct function here as it escapes columns/tables with double quotes `"` but the JSON path requires them to be escaped as values using single quotes `'`.

Reply:
ooh ok! but as discussed, we will not escape them with single quotes but instead move the values in the list of parameters 👍

Reply:
Not sure what "the list of parameters" is but if that escapes values using a single quote then it should be fine 😄

Reply:
the list of parameters stores all user defined values and is passed to the database along with the query. that way we can do parameterized queries to protect against SQL injections.

so up until now the user defined paths were started directly in the query. now I changed that, so that the path values are passed separately (as items in the list of parameters). In the query we now reference a specific value by provided the index of the values from the list of parameters :) so even more secure that just escaping the values 🎉